### ISSUE TITLE
Meshgraphnets : How is wind direction given to the network

### ISSUE THREAD

**Issue by sharma-n** [deepmind/deepmind-research#348]
Hi @alvarosg, @tobiaspfaff,
Firstly, thanks for putting up this work on a public repo! really appreciate it :)
I wanted to understand how for MeshGraphNets, in the FLAG_SIMPLE/FLAG_DYNAMIC cases, the network was given information about the direction and strength of the wind? My current understanding is that the information is somehow embedded in the initial velocities of the graph? Is that correct?

If that is the case, can you explain how you tackled the cases where the wind direction and strength change in the middle of a simulation? does a pure physics simulation software have to come into the loop at that time?

Thanks in advance for any info!

**Comment by alvarosg** [comment]
Thanks for your message!

> I wanted to understand how for MeshGraphNets, in the FLAG_SIMPLE/FLAG_DYNAMIC cases

so in both those cases the wind always goes in the same direction for the whole dataset, so there is not need to give it as a feature to the model, because it is just a bias that the model can learn.

> If that is the case, can you explain how you tackled the cases where the wind direction and strength change in the middle of a simulation?

There is a bit more info on this at the bottom of "Section 5 - Generalization" and bottom of "Section A.1"

For this experiment, we trained the model on a variation of the FLAG_DYNAMIC dataset for which wind speeds and directions actually varied between trajectories (but the same within each full trajectory). In this case, the global wind speed vector is simply appended to the node features at every timestep, and at test time, we simply allowed this value to change in the middle of the trajectory. Unfortunately, to keep it simple, we decided not to include this variant of the dataset as part of the open source process.

Hope this helps.
